clientconn.go

```diff
@@ -362,24 +357,22 @@ func DialContext(ctx context.Context, target string, opts ...DialOption) (conn *
 				return
 			}
 			ch := cc.dopts.balancer.Notify()
-			if ch == nil {
-				// There is no name resolver installed.
-				addrs = append(addrs, Address{Addr: target})
-			} else {
-				addrs, ok = <-ch
-				if !ok || len(addrs) == 0 {
-					waitC <- errNoAddr
-					return
+			if ch != nil {
+				if cc.dopts.block {
+					doneChan := make(chan struct{})
+					go cc.lbWatcher(doneChan)
```

Comment:
In blocking mode, we need a mechanism for lbwatcher to send an error back(if there's one) to DialContext so that the users can be intimated about it.
Also, we need to discuss which error should lbwatcher return since there could be multiple errors while connecting to multiple servers